Zend/zend_object_handlers.c

```diff
@@ -819,7 +819,20 @@ ZEND_API zval *zend_std_write_property(zend_object *zobj, zend_string *name, zva
 				}
 
 				ZVAL_COPY_VALUE(&tmp, value);
-				if (UNEXPECTED(!zend_verify_property_type(prop_info, &tmp, property_uses_strict_types()))) {
+				// Increase refcount to prevent object from being released in __toString()
+				GC_ADDREF(zobj);
+				bool type_matched = zend_verify_property_type(prop_info, &tmp, property_uses_strict_types());
+				if (UNEXPECTED(GC_DELREF(zobj) == 0)) {
+					zend_throw_error(NULL, "Object was released while assigning property %s::$%s",
+						ZSTR_VAL(prop_info->ce->name), zend_get_unmangled_property_name(prop_info->name));
+					zend_objects_store_del(zobj);
+					zval_ptr_dtor(&tmp);
+					variable_ptr = &EG(error_zval);
+					goto exit;
+				} else if (UNEXPECTED(GC_MAY_LEAK((zend_refcounted *)zobj))) {
+					gc_possible_root((zend_refcounted *)zobj);
```

Comment:
I think, we may avoid GC recording here.

Reply:
I think I only added this after experiencing leaks, but I'll try again :slightly_smiling_face:

Reply:
You were right of course, I was under the assumption that `GC_ADDREF` removes the object from the GC buffer but it doesn't.